test/new-e2e/pkg/utils/e2e/client/connection_initializer.go

```diff
@@ -0,0 +1,39 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package client
+
+import (
+	"fmt"
+	"github.com/DataDog/test-infra-definitions/common/utils"
+	"reflect"
+	"testing"
+)
+
+// connectionInitializer defines a method which is used to initialize an connection between
+// testing infrastructure and clients
+type connectionInitializer interface {
+	// InitFromConnection initializes the instance from ssh connection data.
+	// This method is called by [CallConnectionInitializers] using reflection.
+	initFromConnection(t *testing.T, conns map[string]*utils.Connection) error
+}
```

Comment:
I'd like to note that the addition of this interface (and the related additions to `pulumi_stack_initializer.go`) are only required because the `stackInitializer` interface is very closely tied to Pulumi (specifically, when test VMs are provisioned by pulumi their ssh connection information is serialized into/deserialized from an `UpResult` object).

If we were to change the serialization of ssh connection information to make it more generic, then both pulumi & non-pulumi infrastructure could be initialized the same way.